server/src/main/java/org/jboss/as/server/ApplicationServerService.java

```diff
@@ -227,14 +227,12 @@ private static AbstractVaultReader loadVaultReaderService() {
         final Iterator<AbstractVaultReader> it = serviceLoader.iterator();
         // TODO WFCORE-114 get rid of catching/suppressing errors once we have a complete impl in WFCORE
         ServiceConfigurationError sce = null;
-        while (it.hasNext()) {
-            try {
+        try {
+            while (it.hasNext()) {
                 return it.next();
-            } catch (ServiceConfigurationError e) {
-                if (sce == null) {
-                    sce = e;
-                }
             }
+        } catch (ServiceConfigurationError e) {
+            sce = e;
```

Comment:
Same question here as to if this should be done. This will log the last error instead of the first now.